partiql-conformance-tests/build.rs

```diff
@@ -0,0 +1,296 @@
+
+    let mut scope = Scope::new();
+
+    for ion_data in all_ion_data {
+        match ion_data.ion_type() {
+            IonType::Struct => {
+                let test_struct = ion_data.as_struct().unwrap();
+                let test_name = test_struct
+                    .get("name")
+                    .unwrap()
+                    .as_str()
+                    .unwrap()
+                    .to_owned();
+                let statement = test_struct
+                    .get("statement")
+                    .unwrap()
+                    .as_str()
+                    .unwrap()
+                    .to_owned();
+
+                let test_case = ParsePassTestCase {
+                    test_name,
+                    statement,
+                    namespace: None,
+                };
+                scope.push_fn(test_case.to_test_function());
+            }
+            IonType::List => {
+                let test_list = ion_data.as_sequence().unwrap();
+                for tc in test_list.iter() {
+                    match tc.ion_type() {
+                        IonType::Struct => {
+                            let namespace = Some(
+                                ion_data
+                                    .annotations()
+                                    .next()
+                                    .unwrap()
+                                    .text()
+                                    .unwrap()
+                                    .to_owned(),
+                            );
+
+                            let test_struct = tc.as_struct().unwrap();
+                            let test_name = test_struct
+                                .get("name")
+                                .unwrap()
+                                .as_str()
+                                .unwrap()
+                                .to_owned();
+                            let statement = test_struct
+                                .get("statement")
+                                .unwrap()
+                                .as_str()
+                                .unwrap()
+                                .to_owned();
+                            let test_case = ParsePassTestCase {
+                                test_name,
+                                statement,
+                                namespace,
+                            };
+                            scope.push_fn(test_case.to_test_function());
+                        }
+                        _ => panic!("Expected tests within a namespace to be a struct"),
+                    }
+                }
+            }
+            _ => panic!("Unexpected Ion type received"),
+        };
+    }
+    scope
+}
+
+/// Parses the given .ion file `path` and creates a new [`codegen::Scope`] with a
+/// [`codegen::Function`] for each test in `path`.
+///
+/// Assumes the given tests in `path` give errors when parsed.
+fn create_fail_parse_scope(path: &Path) -> Scope {
```

Comment:
There's a lot of repetition here with `create_pass_parse_scope`. I can reduce a lot by changing the modeling of the `ParsePassTestCase`/`ParseFailTestCase` and modifying the function signatures.